### ISSUE TITLE
Unstable validation loss

### ISSUE THREAD

**Issue by chanmuzi** [chanmuzi/NLP-with-Disaster-Tweets#1]
The problems here,
```python
for idx,batch in enumerate(pbar):
      valid_step += 1

      valid_input_ids = batch['input_ids'].to(device)
      valid_attention_mask = batch['attention_mask'].to(device)
      valid_labels = batch['labels'].squeeze().to(device).long()

      logits = model(valid_input_ids, valid_attention_mask).to(device)
      predictions = torch.argmax(logits, dim=1)
      detached_predictions = predictions.detach().cpu().numpy()
      
      loss = loss_fn(logits, valid_labels)
      valid_loss += loss.detach().cpu().numpy().item()

      y_pred.extend(predictions.cpu().numpy())
      y_true.extend(valid_labels.cpu().numpy())

      wandb.log({
          'epoch':epoch,
          'valid_loss':valid_loss/valid_step
      })                

  valid_loss /= valid_step
  f1 = f1_score(y_true,y_pred)

  print(f'Epoch [{epoch+1}/{epochs}] Score: {f1}')
  print(f'Epoch [{epoch+1}/{epochs}] Valid_loss: {valid_loss}')

  if valid_loss < min(valid_loss_min):
      print('model improved!')
  else:
      print('model not improved')

  torch.save(model.state_dict(), f'save/epoch:{epoch+1}_model.pt')
  print('save checkpoint!')
  valid_loss_min.append(valid_loss)
  print(f'valid_loss_min:{min(valid_loss_min)}')
```

![image](https://user-images.githubusercontent.com/101971295/222780199-e5574313-61ce-4723-98fb-69afd0178940.png)
As I know, there are lots of methods recording log during the training or validation.
But, when I tried to record a log during the validation especially each batch trough `wandb.log`, it looks like it has a problem.
So I will try to relocate it to outside the batch, and record logs each epoch.

**Comment by chanmuzi** [comment]
Modified codes are below.
```python
    y_pred.extend(predictions.cpu().numpy())
    y_true.extend(valid_labels.cpu().numpy())

wandb.log({
    'epoch':epoch,
    'valid_loss':valid_loss/valid_step
})
```

I put the wandb.log function outside the batch, and it worked well!
<img width="607" alt="image" src="https://user-images.githubusercontent.com/101971295/222784294-930dac57-29ea-468f-83a4-735db12dea6c.png">
